src/api/auth/auth.module.ts

```diff
@@ -1,13 +1,32 @@
 import { Module } from '@nestjs/common';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { JwtModule } from '@nestjs/jwt';
+import { PassportModule } from '@nestjs/passport';
+import { TypeOrmModule } from '@nestjs/typeorm';
 
 import { UserModule } from '@api/user/user.module';
 
-import { AuthService } from './auth.service';
 import { AuthController } from './auth.controller';
+import { AuthEntity } from './auth.entity';
+import { AuthService } from './auth.service';
+import { JwtStrategy } from './strategies/jwt.strategy';
+import { LocalStrategy } from './strategies/local.strategy';
 
 @Module({
-  imports: [UserModule],
-  providers: [AuthService],
+  imports: [
+    UserModule,
+    PassportModule,
+    JwtModule.registerAsync({
+      imports: [ConfigModule],
+      useFactory: async (config: ConfigService) => ({
+        secret: config.getOrThrow('JWT_SECRET'),
```

Comment:
I agree, although I forgot to put the type `Environment` in the configService which already has all the environment variables defined so it might be more `type-safe` this solution what do you think?